src/plugins/data/public/search/session/session_service.ts

```diff
@@ -56,40 +57,54 @@ export class SessionService {
   private readonly state: SessionStateContainer<TrackSearchDescriptor>;
 
   private searchSessionInfoProvider?: SearchSessionInfoProvider;
-  private appChangeSubscription$?: Subscription;
+  private subs: Subscription[] = [];
```

Comment:
Nit: Can't you just use a parent `Subscription` here that calls `add` with the `sessionStartTime$` and `currentAppId$` subscriptions, then just calls `unsubscribe()` to unsubscribe all its children?